### ISSUE TITLE
FileType request responses randomly "lost" when downloading media

### ISSUE THREAD

**Issue by Lonami** [LonamiWebs/Telethon#658]
Several [`FileType`](https://lonamiwebs.github.io/Telethon/types/storage/file_type.html) constructors (such as [`FileUnknown`](https://lonamiwebs.github.io/Telethon/constructors/storage/file_unknown.html) and [`FileJpeg`](https://lonamiwebs.github.io/Telethon/constructors/storage/file_jpeg.html)) are being "randomly lost" when downloading a lot of media (see https://github.com/expectocode/telegram-export/issues/22 to reproduce the bug, as of the [200th commit](https://github.com/expectocode/telegram-export/commit/7cd76497c454464652467accc3c5b64cd541ffae) and Telethon v0.17.4).

Basically the following message is pretty much spammed when `logging` is enabled:
```
2018-03-01 23:46:26,451 - telethon.network.mtproto_sender - WARNING - Lost request (ID 6528126892703884000) with code 0x96a18d5 will be skipped, contents: FileJpeg()
```

**Comment by Lonami** [comment]
Here's the stripped log:
<details>
<summary>Log:</summary>
<pre>
>> sending 2 6528274468228086776 dict(msg_id=6528274468228086776, seq_no=7, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=262144, limit=262144), container_msg_id=None)
< 6528274468678981633 is rpc result
rpc response 0x96a18d5 for 6528274468228086776 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=262144, limit=262144)
>> sending 2 6528274473251905392 dict(msg_id=6528274473251905392, seq_no=81, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274473461305345 is rpc result
rpc response 0x96a18d5 for 6528274473251905392 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274475330613684 dict(msg_id=6528274475330613684, seq_no=83, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274475361042433 is rpc result
rpc response 0x96a18d5 for 6528274475330613684 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274479878567888 dict(msg_id=6528274479878567888, seq_no=85, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274479933292545 is rpc result
rpc response 0x96a18d5 for 6528274479878567888 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274480995368196 dict(msg_id=6528274480995368196, seq_no=87, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274481191914497 is rpc result
rpc response 0x96a18d5 for 6528274480995368196 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274485638014628 dict(msg_id=6528274485638014628, seq_no=89, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274485840014337 is rpc result
rpc response 0x96a18d5 for 6528274485638014628 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274487738453700 dict(msg_id=6528274487738453700, seq_no=91, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274488063515649 is rpc result
rpc response 0x96a18d5 for 6528274487738453700 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274492608965952 dict(msg_id=6528274492608965952, seq_no=93, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274492656912385 is rpc result
rpc response 0x96a18d5 for 6528274492608965952 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274495249993404 dict(msg_id=6528274495249993404, seq_no=95, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274495512797185 is rpc result
rpc response 0x96a18d5 for 6528274495249993404 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274499814597808 dict(msg_id=6528274499814597808, seq_no=97, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274500093064193 is rpc result
rpc response 0x96a18d5 for 6528274499814597808 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274502184105240 dict(msg_id=6528274502184105240, seq_no=99, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274502286971905 is rpc result
rpc response 0x96a18d5 for 6528274502184105240 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274506737823452 dict(msg_id=6528274506737823452, seq_no=101, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274506873962497 is rpc result
rpc response 0x96a18d5 for 6528274506737823452 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274507910114256 dict(msg_id=6528274507910114256, seq_no=103, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274508142056449 is rpc result
rpc response 0x96a18d5 for 6528274507910114256 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274512531106560 dict(msg_id=6528274512531106560, seq_no=105, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274512783240193 is rpc result
rpc response 0x96a18d5 for 6528274512531106560 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274513364422408 dict(msg_id=6528274513364422408, seq_no=107, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274513740610561 is rpc result
rpc response 0x96a18d5 for 6528274513364422408 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274518276156280 dict(msg_id=6528274518276156280, seq_no=109, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072), container_msg_id=None)
< 6528274518358611969 is rpc result
rpc response 0x96a18d5 for 6528274518276156280 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=131072)
>> sending 2 6528274522526562192 dict(msg_id=6528274522526562192, seq_no=111, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072), container_msg_id=None)
< 6528274522554559489 is rpc result
rpc response 0x96a18d5 for 6528274522526562192 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=131072, limit=131072)
>> sending 2 6528274523596034504 dict(msg_id=6528274523596034504, seq_no=113, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=262144, limit=131072), container_msg_id=None)
< 6528274523703197697 is rpc result
rpc response 0x96a18d5 for 6528274523596034504 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=262144, limit=131072)
>> sending 2 6528274524717418168 dict(msg_id=6528274524717418168, seq_no=115, request=GetHistoryRequest(peer=InputPeerUser(user_id=0, access_hash=0), offset_id=2905712, offset_date=datetime.utcfromtimestamp(1519904241), add_offset=0, limit=100, max_id=0, min_id=0, hash=0), container_msg_id=None)
< 6528274525157910529 is rpc result
rpc response 0x3072cfa1 for 6528274524717418168 has matching req? GetHistoryRequest(peer=InputPeerUser(user_id=0, access_hash=0), offset_id=2905712, offset_date=datetime.utcfromtimestamp(1519904241), add_offset=0, limit=100, max_id=0, min_id=0, hash=0)
>> sending 2 6528274525726490472 dict(msg_id=6528274525726490472, seq_no=117, request=GetFullUserRequest(id=InputUser(user_id=0, access_hash=0)), container_msg_id=None)
< 6528274526052616193 is rpc result
rpc response 0x3072cfa1 for 6528274525726490472 has matching req? GetFullUserRequest(id=InputUser(user_id=0, access_hash=0))
>> sending 2 6528274526059302784 dict(msg_id=6528274526059302784, seq_no=119, request=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=262144), container_msg_id=None)
< 6528274526342187009 is rpc result
rpc response 0x2144ca19 for 6528274526059302784 has matching req? GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=262144)
>> sending 1 6528274527222018340 dict(msg_id=6528274527222018340, seq_no=9, request=InvokeWithLayerRequest(layer=74, query=InitConnectionRequest(api_id=6, device_model='Linux', system_version='4.15.3-2-ARCH', app_version='0.17.4', system_lang_code='en', lang_pack='', lang_code='en', query=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=262144))), container_msg_id=None)
< 6528274527442196481 is container
< processing container inner id 6528274468678981633
< 6528274468678981633 is rpc result
rpc response 0x96a18d5 for 6528274468228086776 has matching req? None
!!!!!!!!!!! LOST !!!!!!!!!!! 6528274468228086776 0x96a18d5 FileJpeg()
< processing container inner id 6528274527442190337
< 6528274527442190337 is MsgsAck(msg_ids=[6528274527222018340])
</pre>
</details>

As it can be seen the server responds twice to the `6528274468228086776` RPC. Strangely enough the:
```python
InvokeWithLayerRequest(layer=74, query=InitConnectionRequest(api_id=6, device_model='Linux', system_version='4.15.3-2-ARCH', app_version='0.17.4', system_lang_code='en', lang_pack='', lang_code='en', query=GetFileRequest(location=InputFileLocation(volume_id=0, local_id=0, secret=0), offset=0, limit=262144)))
```

Seems to happen every time before this is triggered.
